pkg/cmd/roachprod/main.go

```diff
@@ -668,6 +668,36 @@ non-terminating signal (e.g. SIGHUP), unless you also configure --max-wait.
 	}),
 }
 
+var deployCmd = &cobra.Command{
+	Use:   "deploy <cluster> <application> <version>",
+	Short: "deploy a new version of cockroach",
+	Long: `Performs a rolling update, to avoid down time, of cockroach.
+
+The deploy command currently only supports redeploying the storage cluster
+(system tenant). It should be run on a cluster that is already running
+cockroach. The command will download the specified version of cockroach and
+stage it on the cluster. It will then perform a rolling update of the cluster,
+one node at a time, to the new version.
+
+Currently available application options are:
+  cockroach  - Cockroach nightly builds. Can provide an optional SHA, otherwise
+               latest build version is used.
+  release    - Official CockroachDB Release. Must provide a specific release
+               version.
+  customized - Cockroach customized builds, usually generated by running
+               ./scripts/tag-custom-build.sh. Must provide a specific tag.
```

Comment:
Should we extract this and reuse across `stage` and `deploy`?

Reply:
We could, with the exception of `workload`, which is also why I had separate validation.